app/services/ad_renewal_letters_export_service.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require_relative "concerns/can_load_file_to_aws"
+
+class AdRenewalLettersExportService < ::WasteExemptionsEngine::BaseService
+  include CanLoadFileToAws
+
+  def run
+    File.open(file_path, "w:ASCII-8BIT") do |file|
+      file.write(RenewalLettersBulkPdfService.run(ad_expiring_registrations))
+    end
+
+    load_file_to_aws_bucket
+
+    record_content_created
+  rescue StandardError => e
+    Airbrake.notify e, file_name: file_name
+    Rails.logger.error "Generate AD renewal letters export error for #{file_name}:\n#{e}"
+  ensure
+    File.unlink(file_path) if File.exist?(file_name)
+  end
+
+  private
+
+  def file_path
+    Rails.root.join("tmp/#{file_name}")
+  end
+
+  def file_name
+    date = Date.today.to_formatted_s(:plain_year_month_day)
+
+    "ad_renewal_letters_#{date}.pdf"
+  end
+
+  def ad_expiring_registrations
+    WasteExemptionsEngine::Registration
+      .where(contact_email: "waste-exemptions@environment-agency.gov.uk")
+      .where(
+        id: WasteExemptionsEngine::RegistrationExemption
+              .all_active_exemptions
+              .where(expires_on: ad_letters_expires_on)
+              .select(:registration_id)
+      )
+  end
+
+  def bucket_name
+    WasteExemptionsBackOffice::Application.config.letters_export_bucket_name
+  end
+
+  def ad_letters_expires_on
+    WasteExemptionsBackOffice::Application.config.ad_letters_exports_expires_in.days.from_now.to_date
+  end
+
+  def record_content_created
+    WasteExemptionsEngine::AdRenewalLettersExport.create!(
+      expires_on: ad_letters_expires_on,
+      number_of_letters: ad_expiring_registrations.count,
```

Comment:
Yes, I did though about this scenario. The problem is that we can't generate the letters separately outside of the view with the current code. I was thinking one way to make this check would be to check that the presenter can respond without error to every public interface method once initialized with a registration, and when it cannot then we decide that is a failed generation. Still to decide is how we record this "missing/broken" letter in the DB so that we can show the info to the user. 
Anyway, I don't find this scenario important enough and I think we can tackle this problem separately and even release it later on

Reply:
Then I think we need to drop handling this scenario at the moment. The number of letters needs to represent the number successfully generated and included in the PDF, not just the number expiring.

So if we are not ready to handle recording partial successes, then we need to stop and cancel the whole job and record it as failed when one blows up. We can then look to bring back in partial generated letters as part of the new story we've raised.

Reply:
Oh. This makes me feel bad really :/ 
So, 2 questions:
 - Why is it so important that the number there is exactly the number of Letters in the PDF?  
 - After looking at the wireframe, what is the whole point of showing the user a `Partial fail` without telling them which letters have failed? Do we really need that?

Reply:
Started writing some glorious reasons as to why we've got this decision, only to then...

![pivot](https://user-images.githubusercontent.com/1789650/64687897-3d8d8580-d483-11e9-98f4-e31251e6a6ac.gif)

The thing I've come to see is if we leave the number as what ***should*** have been exported, NCCC will then be able to check for themselves based on the generated PDF whether it matches up.

Ideally the service should be able to tell them something didn't export, rather than having to manually tally the PDF to the figure on screen. However we're all agreed that will come as a later story.

So the main thing I've pivoted on is that there is more value in having **Number of letters** represent how many were eligible, instead of just how many were printed (as we can determine that number from the PDF).